system-parachains/coretime/coretime-kusama/src/coretime.rs

```diff
@@ -183,15 +227,27 @@ impl CoretimeInterface for CoretimeAllocator {
 		}
 	}
 
-	fn check_notify_revenue_info() -> Option<(RCBlockNumberOf<Self>, Self::Balance)> {
-		let revenue = CoretimeRevenue::get();
-		CoretimeRevenue::set(&None);
-		revenue
-	}
+	fn on_new_timeslice(t: pallet_broker::Timeslice) {
+		// Burn roughly once per day. Unchecked math: RHS hardcoded as non-zero.
+		if t % 180 != 0 {
+			return
+		}
 
-	#[cfg(feature = "runtime-benchmarks")]
-	fn ensure_notify_revenue_info(when: RCBlockNumberOf<Self>, revenue: Self::Balance) {
-		CoretimeRevenue::set(&Some((when, revenue)));
+		let stash = CoretimeBurnAccount::get();
+		let value =
+			Balances::reducible_balance(&stash, Preservation::Expendable, Fortitude::Polite);
```

Comment:
Drawing other reviewers' attention here after discussion in https://github.com/polkadot-fellows/runtimes/pull/384#discussion_r1677730618

Discussion about keeping the ED vs burning it each time

Reply:
I would keep it, just makes it better if small amounts are deposited later.